neuralop/layers/integral_transform.py

```diff
@@ -90,36 +89,33 @@ def __init__(
     the same as the channels of f
     """
 
-    def forward(
-        self,
-        y,
-        neighbors,
-        x=None,
-        f_y=None,
-        weights=None,
-    ):
+    def forward(self, y, neighbors, x=None, f_y=None, weights=None):
         """Compute a kernel integral transform
 
         Parameters
         ----------
         y : torch.Tensor of shape [n, d1]
             n points of dimension d1 specifying
             the space to integrate over.
+            If batched, these must remain constant
+            over the whole batch so no batch dim is needed.
         neighbors : dict
             The sets A(x) given in CRS format. The
             dict must contain the keys "neighbors_index"
             and "neighbors_row_splits." For descriptions
             of the two, see NeighborSearch.
+            If batch > 1, the neighbors must be constant
+            across the entire batch.
         x : torch.Tensor of shape [m, d2], default None
             m points of dimension d2 over which the
             output function is defined. If None,
             x = y.
-        f_y : torch.Tensor of shape [n, d3], default None
+        f_y : torch.Tensor of shape [batch, n, d3] or [n, d3], default None
             Function to integrate the kernel against defined
             on the points y. The kernel is assumed diagonal
             hence its output shape must be d3 for the transforms
             (b) or (d). If None, (a) is computed.
-        weights : torch.Tensor of shape [n,], default None
+        weights : torch.Tensor of shape [batch, n] or [n,], default None
```

Comment:
What happens if weights have a batch-dim., but there is no `f_y` or it does not have a batch-dim.? It is also unclear whether there is a use-case where we need different weights for different samples in the batch (since they have the same mesh).

Reply:
That's a good point about the batch. Let me add a test to verify the correct repeat shape for unbatched

Reply:
On further thought, this PR doesn't use the GNO weights at all - might be better as a separate issue